torch/utils/data/datapipes/datapipe.py

```diff
@@ -154,6 +155,12 @@ def __getstate__(self):
             return IterDataPipe.getstate_hook(self)
         return self.__dict__
 
+    def __setstate__(self, state):
+        """
+        This method definition is necessary so that the hook function in `_typing.py` can work.
+        """
+        self.__dict__ = state
```

Comment:
I realized this is needed for the hook to work. Let me know if there is a better idea.